src/openvpn/pkcs11.c

```diff
@@ -312,7 +312,7 @@ pkcs11_initialize(
 
     pkcs11h_setLogLevel(_pkcs11_msg_openvpn2pkcs11(get_debug_level()));
 
-    if ((rv = pkcs11h_setForkMode(TRUE)) != CKR_OK)
+    if ((rv = pkcs11h_setForkMode(FALSE)) != CKR_OK)
```

Comment:
as far as I remember, openvpn has early initialization then fork as daemon then runs. I expect the first phase to run with safe fork and then only after daemonizing to set unsafe fork.

Reply:
The `possibly_become_deamon()` is call is before the call to `pkcs11_initialize()` (which is in `context_init_1()`. This makes me think the current patch should be fine.

Reply:
Even if `possibly_become_daemon()` was called after initializing the /pkcs11-helper/ library, I believe that setting "safe fork mode" would not be needed because its functionality is never used from multiple processes/threads.